app/scripts/modules/core/src/search/global/GlobalSearch.tsx

```diff
@@ -0,0 +1,409 @@
+                      const params = result.params || {};
+                      const account = result.account || params.account || params.accountId || params.accountName;
+
+                      return (
+                        <li
+                          key={result.id}
+                          className="result"
+                          onClick={() => {
+                            this.hideResults();
+                            ReactGA.event({ category: 'Global Search', action: 'Recent Item Selected' });
+                          }}
+                        >
+                          <UISref to={result.state} params={result.params}>
+                            <a
+                              ref={(ref) => {
+                                if (this.resultRefs[categoryIndex]) {
+                                  this.resultRefs[categoryIndex][index] = ref;
+                                }
+                              }}
+                              onKeyDown={this.navigateResult}
+                            >
+                              <SearchResult displayName={result.displayName} account={account} />
+                            </a>
+                          </UISref>
+                        </li>
+                      );
+                    })
+                  ]))}
+                </ul>
+              );
+            }}
+          />
+        }
+        {!!query && !querying && showSearchResults &&
+          <ul className="dropdown-menu" role="menu">
+            {categories.map((category, categoryIndex) => ([
+              <li key={category.type.id} className="category-heading">
+                <div className="category-heading">{category.type.displayName}</div>
+              </li>,
+              category.results.map((result, index) => (
+                <li key={result.id} className="result">
+                  <a
+                    onKeyDown={this.navigateResult}
+                    onClick={() => {
+                      ReactGA.event({ category: 'Global Search', action: 'Result Selected' });
+                      this.hideResults();
+                      this.clearFilters(result);
+                    }}
+                    ref={(ref) => {
+                      if (this.resultRefs[categoryIndex]) {
+                        this.resultRefs[categoryIndex][index] = ref;
+                      }
+                    }}
+                    // TODO: probably worth moving these (and the href for 'see more results') over to a UISRef at some point
+                    href={result.href}
+                  >
+                    <SearchResult displayName={result.displayName} account={(result as any).account} />
+                  </a>
+                </li>
+              ))
+            ]))}
+            {!!categories.length && [
+              <li key="divider" className="divider"/>,
+              <li key="seeMore" className="result">
+                <a
```

Comment:
this can be a `UISref` too . 
`to="infrastructure" params={{ q: query }}`
It should redirect appropriately

Reply:
👍 i figured/hoped it would be as easy as that, was intentionally avoiding doing that until a follow up PR.

Reply:
Huh. Weirdly this doesn't seem to work. I keep getting `{"$id":0,"type":4,"message":"This transition is invalid","detail":"Could not resolve 'infrastructure' from state ''"}`.

Will have to look more closely at that.

Reply:
oh, maybe its `to="home.infrastructure"`